cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -507,11 +507,20 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
                 }
             }
             return withVoid(zkSetOperations.maybeRollingUpdate(zkDiffs.resource(),
-                    zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved()));
+                    zkAncillaryCmChange || this.clusterCa.certRenewed() || this.clusterCa.certsRemoved(), false));
         }
 
         Future<ReconciliationState> zkScaleUp() {
-            return withVoid(zkSetOperations.scaleUp(namespace, zkCluster.getName(), zkCluster.getReplicas()));
+            int currentReplicas = 0;
+            StatefulSet ss = zkSetOperations.get(namespace, ZookeeperCluster.zookeeperClusterName(name));
+            if (ss != null) {
+                currentReplicas = ss.getSpec().getReplicas();
+            }
+            Future<Integer> result = Future.succeededFuture(currentReplicas);
+            for (int i = currentReplicas + 1; i <= zkCluster.getReplicas(); i++) {
+                result.compose(scaleTo -> zkSetOperations.scaleUp(namespace, zkCluster.getName(), scaleTo + 1));
```

Comment:
I think it's worth a comment to explain why we only scale by 1, rather than scaling up in one go.